contract_compassion/model/contracts.py

```diff
@@ -0,0 +1,646 @@
+            readonly=True, track_visibility='onchange',
+            help=_(" * The 'Draft' status is used when a user is encoding a "
+                   "new and unconfirmed Contract.\n"
+                   " * The 'Waiting' status is used when the Contract is "
+                   " confirmed but the partner has not yet paid.\n"
+                   " * The 'Active' status is used when the contract is "
+                   "confirmed and until it's terminated.\n"
+                   " * The 'Terminated' status is used when a contract is no "
+                   "longer active.\n"
+                   " * The 'Cancelled' status is used when a contract was "
+                   "never paid.")),
+        'is_active': fields.function(
+            _active, string='Contract Active', type='boolean',
+            store={
+                'account.invoice': (_get_contract_from_invoice, ['state'], 50)
+            },
+            help="It indicates that the first invoice has been paid and the "
+                 "contract was activated."),
+        'fully_managed': fields.function(
+            _is_fully_managed, type="boolean", store=True),
+        # Field used for identifying gifts from sponsor (because of bad GP)
+        'num_pol_ga': fields.integer(
+            'Partner Contract Number', required=True
+        ),
+        'frequency': fields.related(
+            'group_id', 'advance_billing_months',
+            type="integer", readonly=True,
+            string=_('Frequency'), store=False),
+        'end_reason': fields.selection(get_ending_reasons, _('End reason'),
+                                       select=True),
+        'end_date': fields.date(
+            _('End date'), readonly=True,
+            track_visibility="onchange"),
+        'origin_id': fields.many2one('recurring.contract.origin', _("Origin"),
+                                     ondelete='restrict',
+                                     track_visibility='onchange'),
+        'channel': fields.selection(_get_channels, string=_("Channel"),
+                                    required=True, readonly=True,
+                                    states={'draft': [('readonly', False)]}),
+        'parent_id': fields.many2one(
+            'recurring.contract', _('Previous sponsorship'),
+            track_visibility='onchange'),
+        'has_mandate': fields.function(
+            _has_mandate, type='boolean', string='Has mandate'),
+        'name': fields.function(_name_get, type='char'),
+        'partner_id': fields.many2one(
+            'res.partner', string=_('Partner'), required=True,
+            readonly=False, states={'terminated': [('readonly', True)]},
+            ondelete='restrict',
+            track_visibility='onchange'),
+        'type': fields.selection([
+            ('S', _('Sponsorship')),
+            ('O', _('Others'))], _('Type'), select=True,
+            readonly=True, track_visibility='onchange'),
+    }
+
+    _defaults = {
+        'type': 'S',

```

Comment:
do we have to set a type by default ? couldn't it be empty to ensure the user will select the correct value (or the correct value is always set depending from which view we create the contract).